src/v/cluster/health_monitor_backend.cc

```diff
@@ -827,4 +827,20 @@ bool health_monitor_backend::does_raft0_have_leader() {
     return _raft0->get_leader_id().has_value();
 }
 
+void health_monitor_backend::reset() {
+    // Explicitly reset or wipe state here instead of calling stop() because
+    // stop() will also close the gate and break the mutex
+
+    // Do not clear _node_callbacks because this is used to notify other
+    // structures, such as the feature manager, of changes in cluster health
+    // Do not change _reports_disk_health because there is no gaurantee that
+    // disk space changed.
+
+    vlog(clusterlog.info, "Resetting Health Monitor Backend");
+    _last_replies.clear();
+    _reports.clear();
+    _status.clear();
+    _last_refresh = ss::lowres_clock::time_point{};
+}
```

Comment:
Also release `_refresh_request`?
`_reports_disk_health `to init state?

Reply:
> Also release `_refresh_request`? 

@VadimPlh 
Looks like we should also call `abort_current_refresh()` before releasing `_refresh_request`. But the abort will set the promise value to `errc::leadership_changed`. We were thinking about passing in a new error code or setting an exception. Do you have a recommendation here?
Discussion here https://github.com/redpanda-data/redpanda/pull/9282#discussion_r1131583793

Reply:
I decided on overloading `abort_current_refresh` to take a `std::error_code` as an argument. Let me know if this is not preferred.